### PR TITLE
1077 error messages

### PR DESCRIPTION
**Pull request by goodviber**
### Context
We improved the content on the ‘Add course’, and edit course flows to make the language clearer and more specific to the options presented on the page.

At the same time, we changed some of the interactions to make adding and editing a course easier.

This ticket is to handle updating the error messages for all the sections when adding/editing courses.
What needs to be done

Update the error messages when adding/editing course sections to align with the prototype
See the examples to the provided design history which contains the messages to use for all the relevant sections

Success

The error messages are updated and consistent with the prototype/design history.

### Changes proposed in this pull request
update error messages to match design history as far as possible
### Guidance to review
Some pages from the design history would require changes to data defaults and validations e.g. SEND, another date, visas
### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by goodviber** [comment]
[Content changes to add/edit flows - Error messages](https://trello.com/c/xaGHrsUp/1077-content-changes-to-add-edit-flows-error-messages)

**Review by darokel — APPROVED** [review]
Looks fine to me. I think at some point we'll need to start moving these errors out into form objects

File: app/views/publish/courses/accredited_body/new.html.erb [hunk 1]
```diff
@@ -5,59 +5,54 @@
...
-        <%= govuk_link_to(t("cancel"), publish_provider_recruitment_cycle_courses_path(@provider.provider_code, @provider.recruitment_cycle_year)) %>
-      </p>
-
-      </div>
-    </div>
+        </div>
+      <% end %>
+    <% end %>
+  <% end %>
+          <p class="govuk-body">
```
Comment (darokel) [inline app/views/publish/courses/accredited_body/new.html.erb:52]: This markup looks a bit off, could we sort out the indentation?

File: app/views/publish/courses/apprenticeship/_form_fields.html.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<%= render "publish/shared/error_wrapper", error_keys: %i[funding_type program_type], data_qa: "course__funding_type" do %>
+
   <h1 class="govuk-heading-l">
```
Comment (darokel) [inline app/views/publish/courses/apprenticeship/_form_fields.html.erb:2]: Same here re: indentation

File: config/locales/en.yml [hunk 3]
```diff
@@ -295,21 +295,21 @@ en:
...
-              course_creation: "^Select at least one subject"
+              course_creation: "^Select a subject"
               duplicate: "^You have already selected this subject. You can only select a subject once"
             modern_languages_subjects:
               select_a_language: "^Select at least one language"
             study_mode:
               blank: "^Select a study mode"
             applications_open_from:
-              blank: "^Select when applications will open and enter the date if applicable"
+              blank: "^Select an applications open date"
```
Comment (darokel) [inline config/locales/en.yml:312]: I don't think we need these `^` characters anymore but that's for another ticket i think

**Review by darokel — COMMENTED** [review]
Looks good. Left a coupe of comments around the markup then we can get this in

**Review by darokel — APPROVED** [review]

**Pull request merged**